dns/types/msgs.go

```diff
@@ -0,0 +1,161 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// Message types for DNS module
+const (
+	TypeMsgPreCommitEntry string = "pre_commit_entry"
+	TypeMsgCommitEntry    string = "commit_entry"
+)
+
+var _ sdk.Msg = MsgPreCommitEntry{}
+
+// MsgPreCommitEntry is the initial registration of a DNS Entry using a commit
+// reveal scheme.
+type MsgPreCommitEntry struct {
+	// Domain is the lookup key for the DNS Entry.
+	Domain Domain `json:"path" yaml:"path"`
+
+	// Hash is the hash of a random nonce and the encoded entry.
+	Hash []byte `json:"hash" yaml:"hash"`
+
+	// owners of the entry, the first account is expected to be the signer
+	// of this message.
+	Owners []sdk.AccAddress `json:"owners" yaml:"owners"`
+}
+
+// NewMsgPreCommitEntry returns a new instance of MsgPreCommitEntry.
+func NewMsgPreCommitEntry(domain Domain, hash []byte, owners []sdk.AccAddress) MsgPreCommitEntry {
+	return MsgPreCommitEntry{
+		Domain: domain,
+		Hash:   hash,
+		Owners: owners,
+	}
+}
+
+// Route implements sdk.Msg
+func (msg MsgPreCommitEntry) Route() string {
+	return RouterKey
+}
+
+// Type implements sdk.Msg
+func (msg MsgPreCommitEntry) Type() string {
+	return TypeMsgPreCommitEntry
+}
+
+// ValidateBasic ensures that hash is not empty and the path is valid.
+func (msg MsgPreCommitEntry) ValidateBasic() error {
+	if len(msg.Hash) == 0 {
+		return sdkerrors.Wrap(ErrInvalidHash, "hash cannot be empty")
+	}
+	if len(msg.Owners) == 0 {
+		return sdkerrors.Wrap(ErrInvalidOwners, "pre commit entry message must have at least one owner")
+	}
+
+	return msg.Domain.ValidateBasic()
+}
+
+// GetSignBytes implements Msg
+func (msg MsgPreCommitEntry) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+// GetSigners implements sdk.Msg
+func (msg MsgPreCommitEntry) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Owners[0]}
+}
+
+// GetDomain returns the path used to map the DNS entry.
+func (msg MsgPreCommitEntry) GetDomain() Domain {
+	return msg.Domain
+}
+
+// GetOwners returns the owners of the future entry.
+func (msg MsgPreCommitEntry) GetOwners() []sdk.AccAddress {
+	return msg.Owners
+}
+
+// GetHash returns the hash of the random nonce + encoded entry
+func (msg MsgPreCommitEntry) GetHash() []byte {
+	return msg.Hash
+}
+
+// MsgCommitEntry is the final registration of a DNS Entry with the
+// revealed random nonce and the unhashed Entry.
+type MsgCommitEntry struct {
+	// Nonce is the random nonce used in the pre-commit message.
+	Nonce uint64 `json:"nonce" yaml:"nonce"`
+
+	// Domain is the lookup key for the DNS Entry.
+	Domain Domain `json:"domain" yaml:"domain"`
+
+	// Entry is the DNS entry being registered.
+	Entry Entry `json:"entry" yaml:"entry"`
```

Comment:
I note you aren't enforcing an incrementing sequence in the keeper, but letting users set the sequence themselves.

Think the keeper should be enforcing and incrementing sequence

Reply:
yup will add